build-system/tasks/e2e/index.js

```diff
@@ -40,7 +40,6 @@ function installPackages_() {
 }
 
 function buildRuntime_() {
-  execOrDie('gulp clean');
   execOrDie('gulp dist --fortesting');
```

Comment:
This might need `--with_shadow`. I wasn't able to get this to work without that flag. Were you?

Reply:
The shadow runtime is always compiled by multi pass `gulp dist`, since it doesn't support `--watch`.

https://github.com/ampproject/amphtml/blob/2b9201c681fa9e38a50979fc667571123c174b1e/build-system/tasks/helpers.js#L177-L186

Future cleanup task: Remove `--with_shadow` and always compile the shadow runtime, unminified or minified.

Reply:
@cvializ yeah I did! Was it failing during set up? Might be worth noting that the amp4email tests stopped working in shadow env with the compiled script, so I opted to skip them.

Reply:
hmm I don't think shadow roots are intended to be used for rendering dynamic mail, so that sounds safe